packages/discovery-react-components/src/utils/document/transformEnrichment.ts

```diff
@@ -6,19 +6,34 @@ import { ENRICHMENTS, getEnrichmentName } from 'components/CIDocument/utils/enri
 import {
   EnrichedHtml,
   Contract,
+  Invoice,
+  PurchaseOrder,
   Attributes,
   Relations,
   Metadata
 } from 'components/CIDocument/types';
 import { Ontology } from './ontology/types';
 
-const modelMapping = {
-  contract: setMetadata,
-  invoice: setAttributesAndRelations,
-  purchase_order: setAttributesAndRelations
+const ontologyModelMapping = {
+  invoice: invoiceOntology,
+  purchase_order: purchaseOrderOntology
 };
 
-function setMetadata(enrichedHtml: Contract): void {
+function getupdatedEnrichment(enrichmentName: string, enrichment: any): EnrichedHtml {
+  if (enrichmentName === ENRICHMENTS.CONTRACT) {
+    enrichment.metadata = setMetadata(enrichment);
+  } else if (
+    enrichmentName === ENRICHMENTS.INVOICE ||
+    enrichmentName === ENRICHMENTS.PURCHASE_ORDER
+  ) {
+    const { attributes, relations } = setAttributesAndRelations(enrichmentName, enrichment);
+    enrichment.attributes = attributes;
+    enrichment.relations = relations;
+  }
+  return enrichment;
+}
```

Comment:
For example, here, why can't we just do:

```javascript
function getUpdatedEnrichment(..) {
    if (enrichmentName in ontologyModelMapping) {
        enrichment = ontologyModelMapping[enrichmentName](enrichment);
    }
    return enrichment;
}
```

That way, all these bits that update the `enrichment` object are handled by the functions defined in the mapping object.